pkg/locspec/locations.go

```diff
@@ -498,13 +498,19 @@ func SubstitutePath(path string, rules [][2]string) string {
 		}
 
 		// Otherwise check if it's a directory prefix.
-		if !strings.HasSuffix(from, separator) {
+		if from != "" && !strings.HasSuffix(from, separator) {
 			from = from + separator
 		}
-		if !strings.HasSuffix(to, separator) {
+		if to != "" && !strings.HasSuffix(to, separator) {
 			to = to + separator
 		}
-		if strings.HasPrefix(path, from) {
+
+		//  If 'from' is a empty string and path is not absolute, prepende 'to'.
```

Comment:
"prepend"

Although the comment just says the exact same thing as the code. How about "Expand relative paths with the specified prefix"?